### PR TITLE
fix(search): correctly split search words on multiple spaces

### LINKED ISSUE #7818: search highlighting causes php warnings and unexpected output
PHP WARNING: "mb_strpos(): Empty delimiter" in file /engine/lib/mb_wrapper.php (line 96)
PHP WARNING: "mb_substr_count(): Empty substring" in file engine/lib/mb_wrapper.php (line 215)

this is caused by a special search query and some requirements in MySQL/Elgg config

- $CONFIG->seach_info["min_chars"] >= 5
- search query "l  p" (eg "l<space><space>p")

the output for highlighting will be "special"
search highlighting

cause: search_remove_ignored_words doesn't remove " " (eg space) as a word.

solution in search_remove_ignored_words add
```php
$words = explode(' ', $query);
// solution
$words = array_filter($words, function($value) {
	return trim($value) !== "";
});
// end solution

$min_chars = $CONFIG->search_info['min_chars'];
```

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #7818

### CODE REVIEW

File: mod/search/start.php [hunk 1]
```diff
@@ -317,7 +317,12 @@ function search_remove_ignored_words($query, $format = 'array') {
 	$query = stripslashes(strip_tags($query));
 	
 	$words = explode(' ', $query);
```
Comment (mrclay) [inline mod/search/start.php]: We really should use preg_split here instead. We can use a pattern and specify not returning empty items. Using explode plus a filter is confusing to read.

**Comment by mrclay** [comment]
This commit summary is misleading. I would expect this to trim the query client-side and not submit if it were empty. This really fixing the case where the query has multiple consecutive spaces, right?

**Comment by jdalsem** [comment]
correct... will update the commit with a new summary

**Comment by jdalsem** [comment]
updated commit message and switch to preg_split

File: mod/search/start.php [hunk 2]
```diff
@@ -316,7 +316,7 @@ function search_remove_ignored_words($query, $format = 'array') {
 	//$query = str_replace(array('"', '-', '+', '~'), '', stripslashes(strip_tags($query)));
 	$query = stripslashes(strip_tags($query));
 	
-	$words = explode(' ', $query);
+	$words = preg_split('/\s+/', $query);
```
Comment (mrclay) [inline mod/search/start.php]: You should also trim() the $query above. Otherwise you need to add PREG_SPLIT_NO_EMPTY.

Comment (jdalsem) [inline mod/search/start.php]: preference for trim or PREG_SPLIT_NO_EMPTY?

Comment (mrclay) [inline mod/search/start.php]: trim() seems more readable.

Comment (jdalsem) [inline mod/search/start.php]: but also adds to the call stack

Comment (mrclay) [inline mod/search/start.php]: reading > microperf.

**Comment by jdalsem** [comment]
added the trim

**Pull request merged**